graphite/graphite.go

```diff
@@ -89,12 +89,21 @@ func (p *Parser) Parse(line string) (influxdb.Point, error) {
 	fieldValues[name] = v
 
 	// Parse timestamp.
-	unixTime, err := strconv.ParseInt(fields[2], 10, 64)
+	//unixTime, err := strconv.ParseInt(fields[2], 10, 64)
+	unixTime, err := strconv.ParseFloat(fields[2], 64)
 	if err != nil {
 		return influxdb.Point{}, err
 	}
 
-	timestamp := time.Unix(0, unixTime*int64(time.Millisecond))
+	var timestamp time.Time
+	// Check if we have fractional seconds
+	if float64(int64(unixTime)) != unixTime {
```

Comment:
You want to use `math.Floor()`. You also don't need the `if` block. Just do:
```
timestamp = time.Unix(int64(unixTime),  int64((unixTime-math.Floor(unixTime)) * float64(time.Second)))
```
because if we don't have fractional seconds, `int64((unixTime-math.Floor(unixTime)) * float64(time.Second))` will be zero anyway.